recipes/default.rb

```diff
@@ -15,14 +15,14 @@
   node['rackspace_users']['data_bag'],
   node['rackspace_users']['data_bag_item']).to_hash.select { |user, user_data| user != 'id' }
 
-node_tags = node['rackspace_users']['node_tags']
+node_groups = node['rackspace_users']['node_groups']
 
 groups = {}
 
 users.each do |username, user_data|
-  ## The action (if defined in the data bag) should be overwritten to 'remove' if the user's node tags and node's tags don't intersect.
-  ## If the user doesn't declare any tags then it is assumed that is present on all nodes.
-  user_data['action'] = 'remove' if user_data['node_tags'] && (user_data['node_tags'] & node_tags).empty?
+  ## The action (if defined in the data bag) should be overwritten to 'remove' if not all the user's node_groups are in node's node_groups.
+  ## If the user doesn't declare node_groups then it is assumed that is present on all nodes.
+  user_data['action'] = 'remove' if user_data['node_groups'] && !(user_data['node_groups'] - node_groups).empty?
```

Comment:
What I said in my earlier update is that we shouldn't support more than one way. I think that 'any' will cover the 80% of cases as I tried to show with the examples. Adding the whole thing in a  template will give the freedom to cover the rest of the case. This also makes things simpler but I do agree that we need to discuss it more and get other opinions as well. Back to techforum?